### ISSUE TITLE
API Resource modification UI is not user friendly

### ISSUE THREAD

**Issue by tishan89** [wso2/product-apim#10575]
### Describe your problem(s)
When adding a parameter under a resource in publisher portal, when we select body as the type second text box changes to content type. But users are supposed to type it. It would be better to provide a dropdown there
<img width="1174" alt="Screenshot 2021-03-30 at 11 07 59" src="https://user-images.githubusercontent.com/1623470/112945156-88e4b300-9151-11eb-9b62-0f98cfd9c7cb.png">

Once we add the parameter as above content type get shown under the label name which is misleading. Also DataType defaults to Object without and notification.
<img width="1155" alt="Screenshot 2021-03-30 at 11 08 24" src="https://user-images.githubusercontent.com/1623470/112945333-c8130400-9151-11eb-8017-97470c17fdbf.png">

**Comment by HiranyaKavishani** [comment]
Issues that have been mentioned are below.
1. Dropdown for body parameter content type 
Here we have allowed user to type the content type as per his/her requirement since there are set of types and providing all of them are not user friendly.

2. Once we add the parameter as above content type get shown under the label name  is misleading
Fixed with above commit

3. DataType defaults to Object without and notification. 
This cannot be reproduced in latest pack. It is mandatory to fill all three fields in order to enable Add button
